andela/wg-silicon
# Ft create user via api 158626098

####  What does this PR do?
- Adds creation of user accounts through ReST API.
- Adds commands to grant, revoke and check right status of a user to create user via REST API
- Adds command to list users add via API

#### Description of Task to be completed?
Third party applications can add new users via the API using `api/v2/register-user/` as the endpoint.
By default they don't have rights to create users via API thus should request special rights from Wger Admin in order to do so.
Wger Admin grant manage users rights with the following commands:-   
- Grant rights - `./manage.py add_user_rest_api ` `username` `--enable`
- Revoke rights - `./manage.py add_user_rest_api` `username` `--disable`
- Check rights status - `./manage.py add_user_rest_api` `username` `--status`
- To list users created by a certain user - `./manage.py list-user-rest-api ` `username` 

####  How should this be manually tested?
Clone and setup the application locally and have it running
Login as `username`= `admin` and `password` = `admin`
Generate an API key that you will use in the Authorizaiton Header as Token in the third party applications such as Postman to send user data in JSON format including at least username and alternatively use wger endpoint UI to test. (see screenshot for full format). 
At first you get an authorization error since by default users don't have rights to create user. To grant the rights run `./manage.py add_user_rest_api user ` `username` `--enable` on the command line 

What are the relevant pivotal tracker stories?
[Create users via REST API](https://www.pivotaltracker.com/story/show/158626098)

### Screenshots
Sample of commands to check status, grant, revoke rights to create users
![image](https://user-images.githubusercontent.com/22454909/42296605-977d888e-8000-11e8-9a7c-9efca5d62ebc.png)

Sample of command to list created users
![image](https://user-images.githubusercontent.com/22454909/42296621-bc193b98-8000-11e8-9a6e-15f5a74f7ef2.png)

#### Sample of screenshorts on wger UI
User without rights
![image](https://user-images.githubusercontent.com/22454909/42296637-f9fd2316-8000-11e8-81ee-7b069296f159.png)
User with rights
![image](https://user-images.githubusercontent.com/22454909/42296648-18c00200-8001-11e8-90c6-25875d322c7e.png)
Format of users created
![image](https://user-images.githubusercontent.com/22454909/42296664-40bf6f34-8001-11e8-8389-894bd5b23ec3.png)
#### Sample of screenshorts on thrird party application (Postman)
![image](https://user-images.githubusercontent.com/22454909/42296680-620226be-8001-11e8-96a5-218320163716.png)
![image](https://user-images.githubusercontent.com/22454909/42296687-6f154408-8001-11e8-8e3b-b371b30f1557.png)
